packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/SimulationController.java

```diff
@@ -349,41 +347,89 @@ public ResponseEntity<Dataset> createFromSimulationResult(
+			);
+			return ResponseEntity.status(HttpStatus.CREATED).body(dataset);
+		} catch (final Exception e) {
+			final String error = String.format("Failed to add simulation %s result as dataset to project %s", id, projectId);
+			log.error(error, e);
+			throw new ResponseStatusException(org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR, error);
+		}
+	}
 
-			// Duplicate the simulation results to a new dataset
-			simulationService.copySimulationResultToDataset(sim.get(), dataset);
-			datasetService.updateAsset(dataset, projectId, permission);
+	@PostMapping("/{id}/create-assets-from-simulation/{project-id}")
+	@Secured(Roles.USER)
+	@Operation(summary = "Create assets from a simulation result, then add them to a project")
+	@ApiResponses(
+		value = {
+			@ApiResponse(
+				responseCode = "201",
+				description = "Assets created and added to project.",
+				content = @Content(
+					mediaType = MediaType.APPLICATION_JSON_VALUE,
+					schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = ProjectAsset.class)
+				)
+			),
+			@ApiResponse(responseCode = "404", description = "Simulation not found", content = @Content),
+			@ApiResponse(
+				responseCode = "500",
+				description = "There was an issue creating the assets or adding them to the project",
+				content = @Content
+			)
+		}
+	)
+	public ResponseEntity<List<TerariumAsset>> createAssetsFromSimulation(
+		@PathVariable("id") final UUID id,
+		@PathVariable("project-id") final UUID projectId,
+		@RequestParam("name") final String name,
+		@RequestParam("simulation-type") final SimulationType simulationType,
+		@RequestParam(name = "asset-id", required = false) final UUID assetId
+	) {
+		final Schema.Permission permission = projectService.checkPermissionCanWrite(
+			currentUserService.get().getId(),
+			projectId
+		);
 
-			// If this is a temporary asset, do not add to project.
-			if (addToProject == false) {
-				return ResponseEntity.status(HttpStatus.CREATED).body(dataset);
+		try {
+			final Optional<Simulation> sim = simulationService.getAsset(id, permission);
+			if (sim.isEmpty()) {
+				return ResponseEntity.notFound().build();
 			}
 
-			// Add the dataset to the project as an asset
-			final Optional<Project> project = projectService.getProject(projectId);
-			if (project.isPresent()) {
-				projectAssetService.createProjectAsset(project.get(), AssetType.DATASET, dataset, permission);
-				// underlying asset does not exist
-				return ResponseEntity.status(HttpStatus.CREATED).body(dataset);
-			} else {
-				log.error("Failed to add the dataset from simulation {} result", id);
-				return ResponseEntity.internalServerError().build();
+			List<TerariumAsset> assets = new ArrayList<>();
+			// Update the intervention policy name if the simulation is an optimization
+			if (simulationType == SimulationType.OPTIMIZATION) {
+				final Optional<InterventionPolicy> interventionPolicy = interventionPolicyService.getAsset(assetId, permission);
```

Comment:
There is an incoming breaking change to Interventions in [this PR](https://github.com/DARPA-ASKEM/terarium/pull/4902). We probably want to wait on merging PR #4902 first before merging this or else it will propagate the breakage

Reply:
I don't think that this PR will need any code changes though for these changes, more so, we just don't want to be saving references to assets that are immediately going to be blown away